cdap-docs/developers-manual/source/advanced/adapters.rst

```diff
@@ -0,0 +1,140 @@
+another format for archiving purposes. Each Stream event in translated as a line in
+the output fileset. Stream format and schema must be given as adapter properties.
+The output schema will add a required ``ts`` field of type long for the timestamp of the
+event, as well as nullable string fields for any headers given in the adapter properties. 
+
+Adapters of the stream-conversion type require the following adapter properties:
+
+  * frequency: how often the Adapter should run, as a number followed by a number. For
+    example, 5m means the Adapter will run every five minutes, and 2h means the Adapter
+    will run every two hours. Note that this must be a number that divides evenly into
+    the time unit given. For example, 30m is valid, but 13m is not because an hour does
+    not evenly divide into 13 minute chunks.
+  * source.name: the name of the Stream to read from.
+  * source.format.name: the name of the format of the Stream body.
+  * source.format.settings: settings of the format as a JSON Object of string to string.
+  * source.schema: the schema of the Stream body.
+  * sink.name: the name of the TimePartitionedFileSet to write to. 
+
+The following adapter properties are optional:
+
+  * mapper.resources.memory.mb: how much memory each mapper of the mapreduce job should use.
+  * mapper.resources.vcores: how many virtual cores each mapper of the mapreduce job should use.
+  * headers: which stream headers should be included in the output. 
+
+The following sink properties are required:
+
+  * explore.table.property.avro.schema.literal: the output schema with timestamp and headers included. 
+
+Each time the Adapter runs, it will read stream events from a time range, write to the fileset, and add those
+files as a partition. The time range is calculated based on the frequency of the Adapter and the logical
+start time of the Adapter run. For example, if the Adapter is configured to run every hour, each run of the
+Adapter will read events starting from the logical start time minus an hour and ending at the logical start time.
+The logical start time is the time at which the schedule is triggered. For an hour frequency, this means
+the logical start time will be on the hour, every hour. Even if the run actually begins at 5:03, the logical
+start will be 5:00 and the run will read Stream events that were added from 4:00 to 4:59.
+
+Because there are so many properties, and because properties like schema must be string values of
+escaped JSON Objects, it is highly recommended that you use the CLI to create stream-conversion adapters.
+For example::
+
+  cdap (http://127.0.0.1:10000)> create stream-conversion adapter convert_tickers on tickers frequency 10m format csv schema "ticker string, trades int, price double"
+
+The command above will create a stream-conversion Adapter named convert_tickers that runs every ten minutes.
+It will read from the ``tickers`` Stream and write to the ``tickers_converted`` TimePartitionedFileSet.
```

Comment:
```
It will read from the *tickers* Stream and write to the *tickers_converted* ``TimePartitionedFileSet``.
It will require stream events to be comma-separated values of ticker, trades, and price, and will write
events as partitioned Avro files.
```